File: app/src/main/java/com/agrogames/islandsofwar/factories/PointFactory.java
package com.agrogames.islandsofwar.factories;

import com.agrogames.islandsofwar.engine.abs.common.Point;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class PointFactory {
    public static Point[] points(JSONArray json) throws JSONException {
        Point[] result = new Point[json.length()];
        for (int i = 0; i < json.length(); i++){
            result[i] = point(json.getJSONObject(i));
        }
        return result;
    }

    public static Point point(JSONObject json) throws JSONException{
        return new Point((float)json.getDouble("x"), (float) json.getDouble("y"));
    }
}

File: app/src/main/java/com/agrogames/islandsofwar/factories/PointFactory.kt
package com.agrogames.islandsofwar.factories

import com.agrogames.islandsofwar.engine.abs.common.Point
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

object PointFactory {
    @Throws(JSONException::class)
    fun points(json: JSONArray): Array<Point?> {
        val result = arrayOfNulls<Point>(json.length())
        for (i in 0 until json.length()) {
            result[i] = point(json.getJSONObject(i))
        }
        return result
    }

    @Throws(JSONException::class)
    fun point(json: JSONObject): Point {
        return Point(json.getDouble("x").toFloat(), json.getDouble("y").toFloat())
    }
}